src/main/java/in/projecteka/consentmanager/clients/PatientServiceClient.java

```diff
@@ -0,0 +1,32 @@
+package in.projecteka.consentmanager.clients;
+
+import in.projecteka.consentmanager.clients.model.LinkedCareContexts;
+import in.projecteka.consentmanager.clients.properties.LinkServiceProperties;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Mono;
+
+import static org.springframework.http.HttpHeaders.AUTHORIZATION;
+
+public class PatientServiceClient {
+
+    private final WebClient.Builder webClientBuilder;
+    private LinkServiceProperties serviceProperties;
+
+    public PatientServiceClient(WebClient.Builder webClientBuilder, LinkServiceProperties serviceProperties) {
+        this.webClientBuilder = webClientBuilder;
+        this.serviceProperties = serviceProperties;
```

Comment:
we could just pass the String url to the constructor, instead of passing the whole properties, and move the properties out of clients.